signbank/dictionary/tasks.py

```diff
@@ -0,0 +1,68 @@
+from urllib.request import urlretrieve
+
+from background_task import background
+from django.conf import settings
+
+from .models import FieldChoice, Gloss
+from ..video.models import GlossVideo
+
+
+def move_glossvideo_to_valid_filepath(glossvideo):
+    """
+    Mimics the rename_file method on GlossVideo without changing the assigned filename.
+
+    This step is necessary because we create the videos in bulk, and usually the filename and path
+    are updated in the save() step.
+    We only want the path updates, changing the filename would lead to loss of unique file names
+    """
+    old_file = glossvideo.videofile
+    full_new_path = glossvideo.videofile.storage.get_valid_name(
+        glossvideo.videofile.name.split("/")[-1]
+    )
+    if not glossvideo.videofile.storage.exists(full_new_path):
+        # Save the file into the new path.
+        saved_file_path = glossvideo.videofile.storage.save(full_new_path, old_file)
+        # Set the actual file path to videofile.
+        glossvideo.videofile = saved_file_path
+    return glossvideo
+
+
+# background task schedule kicks task off 30 seconds after call
+@background(schedule=30)
+def retrieve_videos_for_glosses(video_details, gloss_pks):
```

Comment:
**suggestion**: I know the existing style doesn't use type annotation, but it would be really helpful to have that here